### ISSUE TITLE
App closes after a being connected for a while (USB)

### ISSUE THREAD

**Issue by alex-ui** [dev47apps/droidcam#165]
**Describe the bug**
The app closes itself after a while connected through usb.

**Expected behavior**
It shouldn't.

**Additional context**
Android Oreo
Windows 10 20H2
HD Mode
DroidCamX 6.9.7

**Comment by aramg** [comment]
Does the client crash, or just stop the video?
If its a crash, could you check if C:\ProgramData\droidcam.log has any crash info after it stops.
If its just video, then it could be a USB port/cable issue.

**Comment by nandwani-rohit** [comment]
Same here.

The Android Application closes after a few minutes, and then the connection to the PC Client is lost. I have to connect again and again. I did check for any power saving settings leading to this behavior on my phone. That is not the case. 

And yes, the client does not crash, just the video feed is lost, and connection too.

**Comment by nandwani-rohit** [comment]
> Does the client crash, or just stop the video?
> If its a crash, could you check if C:\ProgramData\droidcam.log has any crash info after it stops.
> If its just video, then it could be a USB port/cable issue.

Same here.

The Android Application closes after a few minutes, and then the connection to the PC Client is lost. I have to connect again and again. I did check for any power saving settings leading to this behavior on my phone. That is not the case. 

And yes, the client does not crash, just the video feed is lost, and connection too.

**Comment by aramg** [comment]
Thanks, this seems to be affecting some users after the last update. Looking into it.
Any chance you can use the About page of DroidCamX to generate and send a bug report. Do it immediately after the crash, hopefully the logs will contain some traces.

**Comment by alex-ui** [comment]
> Does the client crash, or just stop the video?
> If its a crash, could you check if C:\ProgramData\droidcam.log has any crash info after it stops.
> If its just video, then it could be a USB port/cable issue.

The client crashes and, consequently, stops the video source

**Comment by theclyde** [comment]
Experiencing the same - but this is the first time I am using the app and camera and device, so there is no baseline
 USB Borescope connected to ASUS Chromebook.  App finds camera, computer on network finds feed. A few minutes later the app dims, then a few minutes later the app is not running.  Will try to capture logs after work

**Comment by milindgoel15** [comment]
The logs says the activity just dies on its own. So weird

`<<< log_count = 24 >>>
[05-18 12:08:24.406 1893:3094 I/ActivityTaskManager]
START u0 {flg=0x34100000 cmp=com.dev47apps.droidcamx/.DroidCamX} from uid 10496

[05-18 12:08:24.721 1893:2064 I/ActivityTaskManager]
Displayed com.dev47apps.droidcamx/.DroidCamX: +282ms

[05-18 12:08:28.435 1135:3216 I/CameraService]
CameraService::connect call (PID -1 "com.dev47apps.droidcamx", camera ID 0) for HAL version default and Camera API version 1

[05-18 12:08:28.436 1135:3216 I/Camera2ClientBase]
Camera 0: Opened. Client: com.dev47apps.droidcamx (PID 11313, UID 10496)

[05-18 12:08:28.671 1338:1365 E/statsd]
Predicate 5980654721335871649 dropping data for dimension key (10)0x2010101->10496[I] (10)0x30000->droidcamx:wake-lock[S] 

[05-18 12:08:28.906 1893:3595 V/WindowManager]
Orientation start waiting for draw, mDrawState=DRAW_PENDING in Window{d78e19d u0 com.dev47apps.droidcamx/com.dev47apps.droidcamx.DroidCamX}, surfaceController Surface(name=com.dev47apps.droidcamx/com.dev47apps.droidcamx.DroidCamX)/@0xf0ed9a3

[05-18 12:08:28.957 1893:1981 V/WindowManager]
Orientation start waiting for draw, mDrawState=DRAW_PENDING in Window{d78e19d u0 com.dev47apps.droidcamx/com.dev47apps.droidcamx.DroidCamX}, surfaceController Surface(name=com.dev47apps.droidcamx/com.dev47apps.droidcamx.DroidCamX)/@0xf0ed9a3

[05-18 12:12:09.886 1893:5468 I/ActivityTaskManager]
START u0 {cmp=com.dev47apps.droidcamx/.ListSelector (has extras)} from uid 10496

[05-18 12:12:10.012 1893:2064 I/ActivityTaskManager]
Displayed com.dev47apps.droidcamx/.ListSelector: +122ms

[05-18 12:12:19.747 1893:2722 W/InputDispatcher]
channel 'd78e19d com.dev47apps.droidcamx/com.dev47apps.droidcamx.DroidCamX (server)' ~ Consumer closed input channel or an error occurred.  events=0x9

[05-18 12:12:19.748 1893:2722 E/InputDispatcher]
channel 'd78e19d com.dev47apps.droidcamx/com.dev47apps.droidcamx.DroidCamX (server)' ~ Channel is unrecoverably broken and will be disposed!

[05-18 12:12:19.757 1893:6043 D/ConnectivityService]
ConnectivityService NetworkRequestInfo binderDied(NetworkRequest [ LISTEN id=2176, [ Transports: WIFI Capabilities: NOT_RESTRICTED&TRUSTED&NOT_VPN&FOREGROUND Uid: 10496 AdministratorUids: [] RequestorUid: 10496 RequestorPackageName: com.dev47apps.droidcamx] ], android.os.BinderProxy@629c0cf)

[05-18 12:12:19.759 1893:6041 I/ActivityManager]
Process com.dev47apps.droidcamx (pid 11313) has died: fg  TOP 

[05-18 12:12:19.760 1338:1365 E/statsd]
Predicate 5980654721335871649 dropping data for dimension key (10)0x2010101->10496[I] (10)0x30000->droidcamx:wake-lock[S] 

[05-18 12:12:19.761 1893:6043 I/WindowManager]
WIN DEATH: Window{d78e19d u0 com.dev47apps.droidcamx/com.dev47apps.droidcamx.DroidCamX}

[05-18 12:12:19.761 1893:6043 W/InputDispatcher]
Attempted to unregister already unregistered input channel 'd78e19d com.dev47apps.droidcamx/com.dev47apps.droidcamx.DroidCamX (server)'

[05-18 12:12:19.784 1893:6041 W/ActivityManager]
Scheduling restart of crashed service com.dev47apps.droidcamx/com.dev47apps.dc.DroidCamService in 1000ms for start-requested

[05-18 12:12:19.785 1893:6041 W/ActivityTaskManager]
Force removing ActivityRecord{77cc213 u0 com.dev47apps.droidcamx/.DroidCamX t943}: app died, no saved state

[05-18 12:12:20.140 1135:3216 I/Camera2ClientBase]
Closed Camera 0. Client was: com.dev47apps.droidcamx (PID 11313, UID 10496)

[05-18 12:12:20.806 1893:2070 I/ActivityManager]
Start proc 26924:com.dev47apps.droidcamx/u0a496 for service {com.dev47apps.droidcamx/com.dev47apps.dc.DroidCamService}

[05-18 12:12:20.841 26924:26924 E/7apps.droidcam]
Not starting debugger since process cannot load the jdwp agent.

[05-18 12:12:20.851 26924:26924 W/7apps.droidcam]
Core platform API violation: Ljava/lang/reflect/Field;->accessFlags:I from Landroid/os/Build; using reflection

[05-18 12:12:20.884 26924:26924 I/7apps.droidcam]
The ClassLoaderContext is a special shared library.

[05-18 12:12:20.926 26924:26924 I/TetheringManager]
registerTetheringEventCallback:com.dev47apps.droidcamx`

**Comment by aramg** [comment]
```
[05-18 12:12:19.747 1893:2722 W/InputDispatcher]
channel 'd78e19d com.dev47apps.droidcamx/com.dev47apps.droidcamx.DroidCamX (server)' ~ Consumer closed input channel or an error occurred. events=0x9

[05-18 12:12:19.748 1893:2722 E/InputDispatcher]
channel 'd78e19d com.dev47apps.droidcamx/com.dev47apps.droidcamx.DroidCamX (server)' ~ Channel is unrecoverably broken and will be disposed!
```

There are other similar posts online, people have also encountered this error but no clear solution.

Is this from Android 6.0.1? (all reports so far have been Android 6.0.1)
Does it help if you turn off Auto-Dim in the apps Settings?

**Comment by milindgoel15** [comment]
I have android 11 

> Is this from Android 6.0.1? (all reports so far have been Android 6.0.1)

I have never tried it. But will give it a try today

> Does it help if you turn off Auto-Dim in the apps Settings?

**Comment by aramg** [comment]
I will see if I can reproduce this on one of my phones.

Does DroidCam OBS work without crashing?

**Comment by milindgoel15** [comment]
Yep
> Does DroidCam OBS work without crashing?

**Comment by aramg** [comment]
Just got a tip that the "Keep Device Awake" option may be the cause.

**Comment by VasantJ3** [comment]
Anyone found any solution to this issue. I am still facing this issue. I tried using a different mobile phone, still, the issue persists. After few minutes of using, the Video feed stops and we have to restart the connection and app.

**Comment by milindgoel15** [comment]
Any update on this?

**Comment by aramg** [comment]
Unfortunately the cause is still a mystery. There havent been any actionable crash reports, so far this seems like a system issue that somehow needs to be worked around.
@VasantJ3 which mobile phones is this happening for you? Make/Mode/Android OS version.

**Comment by milindgoel15** [comment]
It seems only happen on pro version. Free version and the obs version has no issues like that. Maybe make a test build from it with premium features included and let your testers test it.

**Comment by StephanBe** [comment]
I have the same issue on an old Zenfone 2 (ASUS_Z00AD with Android 5.0) with the latest DroidCam app.
Sometimes the connection is lost, but the app remains. Sometimes the app crashes back to home screen.

Things that did not help:
- Developer option for keeping the device active while charging is on
- Turning off keep-device-awake option in app
- Turning off auto-dim option in app

Is there a log on the Phone I could provide for further investigation?

**Comment by aramg** [comment]
Seems to consternated to ASUS. 
@StephanBe after connecting over USB and the app closing unexpectedly, you can open a Command Prompt (right-click the Start Menu) and enter the following commands:
```
cd "C:\Program Files (x86)\DroidCam\adb"
adb logcat -t 5000 > "C:\ProgramData\logcat.txt"
```

Email C:\ProgramData\logcat.txt to dev47apps@gmail.com

(on Linux its just: `adb logcat -t 5000 > /tmp/logcat.txt`)

**Comment by StephanBe** [comment]
> Seems to consternated to ASUS.
> @StephanBe after connecting over USB and the app closing unexpectedly, you can open a Command Prompt (right-click the Start Menu) and enter the following commands:
> 
> ```
> cd "C:\Program Files (x86)\DroidCam\adb"
> adb logcat -t 5000 > "C:\ProgramData\logcat.txt"
> ```
> 
> Email C:\ProgramData\logcat.txt to [dev47apps@gmail.com](mailto:dev47apps@gmail.com)
> 
> (on Linux its just: `adb logcat -t 5000 > /tmp/logcat.txt`)

Sent the logcat via email.

**Comment by aramg** [comment]
Attached is an older version of the app, could someone try replacing the current version with it and see if the issue still happens.
https://dev47apps.com/files/apk/droidcamx72.apk

**Comment by StephanBe** [comment]
> Attached is an older version of the app, could someone try replacing the current version with it and see if the issue still happens.
> https://dev47apps.com/files/apk/droidcamx72.apk

Unfortunately this version also crashes after some time. Maybe the camera gets forcibly deactivated from the driver due to supposedly idleness, but this is just an uneducated wild guess.

Uploaded a (censored) logcat of this new crash.
[logcat.txt](https://github.com/dev47apps/droidcam/files/6763079/logcat.txt)

As a clunky workaround: If the DroidCam client would notice the dying connection it should be possible to simply restart the application via adb and reconnect.

**Comment by aramg** [comment]
Auto-reconnections are on the backlog, it's a little tricky because there needs to be a distinction between unexpected drops versus app being stopped, on both ends.

Could you check if the "Bluethooth Link" option is enabled in the app settings, under the microphone section?

Lets also go back a little more, here are some more old versions to test
https://dev47apps.com/files/apk/droidcamx71.apk
https://dev47apps.com/files/apk/droidcamx63.apk

**Comment by StephanBe** [comment]
Version 72 (6.9.4) ran with Bluetooth Link disabled. I retried with enabled link and activated phone Bluetooth, but it still crashed after some time.

Versions 63 (6.8.3) as well as 71 (6.9.3) ran stable for hours in my tests! Even when turning off phone screen or making another app active it does not crash. Though, the video becomes laggy when turning off the screen in all versions.

**Comment by aramg** [comment]
OK, great. Now the question is what broke :)
Here is the latest version, with 2 separate changes that may have been the root cause:
https://dev47apps.com/files/apk/droidcamx77_nol.apk
https://dev47apps.com/files/apk/droidcamx77_ntf.apk

Could we see if either of them work.
The other possible culprit is the bluetooth stack on your devices, I've had nothing but bad experiences with bluetooth over the years.

**Comment by StephanBe** [comment]
Unfortunately both crashed after some time (with bluetooth turned off).

**Comment by aramg** [comment]
OK, thanks 🙏
2 more tweaks:
https://dev47apps.com/files/apk/droidcamx77_ascl.apk
https://dev47apps.com/files/apk/droidcamx77_cln.apk

**Comment by StephanBe** [comment]
Again, both crashed after a few minutes. :disappointed:
Thanks for your effort!

**Comment by aramg** [comment]
@StephanBe 
* Does the crash happen if only video is enabled, or if only audio is enabled?
* Also I believe you're on Linux, does the crash happen if you just curl the app: `curl "PHONE-IP:4747/video" > /dev/null`

@milindgoel15 @VasantJ3 could you confirm if r71 from above works for you too, and r72 crashes.
Would be nice to confirm the cause is exactly between that update.

**Comment by StephanBe** [comment]
> @StephanBe
> 
> * Does the crash happen if only video is enabled, or if only audio is enabled?

The app crashes in both cases.
> * Also I believe you're on Linux, does the crash happen if you just curl the app: `curl "PHONE-IP:4747/video" > /dev/null`

Yes, the crash also happens if I just curl the video via USB.

One thing I noticed is that the app crashes **exactly** after connecting for **5:00 min**, regardless of the generated traffic, even the app versions (as far as I can tell) or me tapping on the phone screen at 4:40 min.

```
$ curl "127.0.0.1:4747/video" > /dev/null 
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  184M    0  184M    0     0   629k      0 --:--:--  0:05:00 --:--:--  620k
$ curl "127.0.0.1:4747/video" > /dev/null 
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  194M    0  194M    0     0   663k      0 --:--:--  0:05:00 --:--:--  661k
$ curl "127.0.0.1:4747/video" > /dev/null 
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  195M    0  195M    0     0   665k      0 --:--:--  0:05:00 --:--:--  624k
$ 
```

**Comment by aramg** [comment]
Here is yet another build, could you curl it and then do a log capture after the crash.
https://dev47apps.com/files/apk/droidcamx77_3.apk

Note that you can generate a log from the app itself (your choice). After the crash go into the "About" page inside DroidCamX and there is a bug report option under (⋮) menu.

**Comment by emanueltimlopez** [comment]
> 
> 
> Here is yet another build, could you curl it and then do a log capture after the crash.
> https://dev47apps.com/files/apk/droidcamx77_3.apk
> 
> Note that you can generate a log from the app itself (your choice). After the crash go into the "About" page inside DroidCamX and there is a bug report option under (⋮) menu.

I'm facing the same issue, pro versión on Samsung A9. This versión works fine for me!

**Comment by aramg** [comment]
@emanueltimlopez ok, great.
Was this with the Linux client, or the Windows client.
Could I get a lo capture of the app, after ~5 minutes of it working. You can stop the video and use the bug report feature in the About page of the app.

**Comment by StephanBe** [comment]
> Here is yet another build, could you curl it and then do a log capture after the crash.
> https://dev47apps.com/files/apk/droidcamx77_3.apk
> 
> Note that you can generate a log from the app itself (your choice). After the crash go into the "About" page inside DroidCamX and there is a bug report option under (⋮) menu.

This version seem to work for me, too. Tested for half an hour.

I sent a logcat via Email from after a bit over 5 min. The last thing logged, should be due to me cancelling the curl stream at 5:04 min of connection.

**Comment by aramg** [comment]
@StephanBe awesome, I think I know what exactly the problem is but need the logs to see how it gets to that point (the thing thats now working should be broken on all devices, its unexpected that only certain devices have the problem).
I think you missed the attachment in the email, could you send the log capture again please 🙏

**Comment by StephanBe** [comment]
> @StephanBe awesome, I think I know what exactly the problem is but need the logs to see how it gets to that point (the thing thats not working should be broken on all devices, its unexpected that only certain devices have the problem).
> I think you missed the attachment in the email, could you send the log capture again please

Ops, I sent the file. :blush:

**Comment by aramg** [comment]
OK all, here is a final build. If we can confirm it works, then a release can be made in the app store
https://dev47apps.com/files/apk/droidcamx77_4.apk

**Comment by StephanBe** [comment]
Unfortunately, this version crashes after exactly 2:40 min.

I sent the corresponding logcat.

**Comment by aramg** [comment]
@StephanBe thanks, check out the reply.

@ Everyone experiencing this problem: do y'all have any security/anti-virus/similar apps installed, that might "analyze" or otherwise manipulate other apps on the phone?

**Comment by StephanBe** [comment]
Update for 77_4: I turned off my "Blokada" and uninstalled it, which was the only thing that could interfere with droidcam. However, the version 77_4 still crashed after some time (3:21 min or 2:42 min, when switching out to other apps, or exactly 2:40 min when keeping the app active).

**Comment by aramg** [comment]
Thanks @StephanBe please also check the e-mail follow up

**Comment by alex-ui** [comment]
> Update for 77_4: I turned off my "Blokada" and uninstalled it, which was the only thing that could interfere with droidcam. However, the version 77_4 still crashed after some time (3:21 min or 2:42 min, when switching out to other apps, or exactly 2:40 min when keeping the app active).

I don't mind if the app closes in the background, but it should not close while having it active and using it (since leaving it in the background disconnects it from the PC).

**Comment by aramg** [comment]
DroidCamX update is being rolled out, the regular app update is waiting for the play store review to clear and will follow.

**Comment by StephanBe** [comment]
The rolled out version works like a charm (at least for my device). Many thanks for the fix. :clap:

**Comment by aramg** [comment]
Awesome, thanks so much for all the help @StephanBe. Contact me anytime if needed.

**Comment by sachingopal** [comment]
Same here. App dies after sometime.

**Comment by aramg** [comment]
Could you email dev47apps@gmail.com.
We'll need to try and diagnose the issue, there hasn't been any uptick in crashes on the Play Store.

**Comment by chromer030** [comment]
This bug ruined one of my most important interviews.

**Comment by alirezaprogrammermaker** [comment]
I have same problem. 
When i connected to pc with usb or wifi app automatically closed.
PHONE MODEL : SAMSUNG GALAXY A71

**Comment by mbarbone1369** [comment]
Hello,
I am having the same problem. I have tried restarting the phone, clear cache, clear data, reinstall, etc, etc, etc....No matter what I do it will not connect again. I tried changing front / back camera, nothing. I did notice something just now. I exited the program again because it wasn't working. I opened Settings and changed Screen Timeout to Never. When I loaded DroidCam this time it took several seconds to load and the interface to appear, but then streaming was functioning again. When I have the no streaming issue, the app opens and displays the interface almost instantly. It's like whatever has killed the actual main process, but the interface is appearing making it look like the app is working. 
DroidCam (Classic) 6.27
Alcatel 5002B
